datadog_checks_dev/datadog_checks/dev/tooling/commands/validate/imports.py

```diff
@@ -0,0 +1,96 @@
+# (C) Datadog, Inc. 2018-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import os
+
+import click
+
+from ...constants import get_root
+from ...utils import complete_valid_checks, get_valid_integrations
+from ..console import CONTEXT_SETTINGS, abort, echo_debug, echo_failure, echo_info, echo_success, echo_warning
+
+
+def validate_import(filepath, check, autofix):
+    """Validate imports are coming from the correct base package."""
+    # almost every case is of the form `from datadog_checks.. import ..`
+    # we want to ensure that the imports are from `datadog_checks.base...`
+    # except for cases where its importing from actual check code
+
+    success = True
+    lines = []
+
+    with open(filepath) as f:
+        for num, line in enumerate(f):
+            if all(
+                (
+                    'import' in line,
+                    'datadog_checks' in line,
+                    'datadog_checks.base' not in line,
+                    f'datadog_checks.{check}' not in line,
+                    'datadog_checks.dev' not in line,
+                )
+            ):
+                success = False
+                lines.append((num, line))
+
+    if autofix and not success:
+        with open(filepath, 'r') as f:
+            data = f.readlines()
+
+        for num, _ in lines:
+            data[num] = data[num].replace('datadog_checks', 'datadog_checks.base')
+
+        with open(filepath, 'w') as f:
+            f.write(''.join(data))
+
+    return success, lines
+
+
+@click.command(context_settings=CONTEXT_SETTINGS, short_help='Validate proper base imports')
+@click.argument('checks', nargs=-1, autocompletion=complete_valid_checks, required=False)
+@click.option('--autofix', is_flag=True, help='Apply suggested fix')
+@click.pass_context
+def imports(ctx, autofix, checks):
+    """Validate proper imports in checks."""
+
+    validation_fails = {}
+    integrations_root = get_root()
+    all_checks = sorted(get_valid_integrations())
+
+    if checks:
+        invalid_checks = [c for c in checks if c not in all_checks]
+        if invalid_checks:
+            abort(f'Invalid checks: {invalid_checks}')
+        all_checks = sorted(checks)
+
+    echo_info("Validating imports avoiding deprecated modules ...")
+    for check_name in sorted(all_checks):
+        echo_debug(f'Checking {check_name}')
+
+        # focus on check and testing directories
+        bases = [os.path.join(integrations_root, check_name, base) for base in ('datadog_checks', 'tests')]
```

Comment:
nit: some of the file/directory logic could be moved to functions in utils

Reply:
Could you clarify this one a bit?  Which functions would be useful in as a utility function?

Reply:
I was thinking something like turning this line into `get_python_files_for_check(check)` or something like that. It could be useful for future scripts

Reply:
This is a good idea - but may be better suited for a separate PR.  There are several options which could be helpful and not sure those discussions would be suited for here.

Reply:
👍